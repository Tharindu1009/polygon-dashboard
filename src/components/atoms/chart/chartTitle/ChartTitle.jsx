
import './styles/chartTitle.scss';
import * as React from 'react';

function ChartTitle({ title }) {
    return (
        <div data-testid="chartTitle" className='chartTitle'>
            {title}
        </div>
    );
}

export default ChartTitle;

