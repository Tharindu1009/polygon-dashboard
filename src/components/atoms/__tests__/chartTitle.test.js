import { render, screen, cleanup } from "@testing-library/react";
import TestRenderer from 'react-test-renderer';
import ChartTitle from "../chart/chartTitle/ChartTitle";

afterEach(()=>{
    cleanup();
})

test('should render ChartTitle component', () => {
    render(<ChartTitle title="Last Open & Close Prices"/>);
    const chartTitleElement = screen.getByTestId('chartTitle');
    expect(chartTitleElement).toBeInTheDocument();
    expect(chartTitleElement).toHaveTextContent('Last Open & Close Prices');
})

test('matches chartTitle snapshot', () => {
    const tree = TestRenderer.create(<ChartTitle title="Last Open & Close Prices"/>).toJSON();
    expect(tree).toMatchSnapshot();
})