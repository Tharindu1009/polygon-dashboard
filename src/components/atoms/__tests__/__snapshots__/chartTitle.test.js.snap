// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches chartTitle snapshot 1`] = `
<div
  className="chartTitle"
  data-testid="chartTitle"
>
  Last Open & Close Prices
</div>
`;

exports[`matches snapshot 1`] = `
<div
  className="chartTitle"
>
  Last Open & Close Prices
</div>
`;

exports[`matches snapshot1 1`] = `
<div
  className="chartTitle"
  data-testid="chartTitle"
>
  Last Open & Close Prices
</div>
`;
